intro-to-rails/routes-and-resources.md

```diff
@@ -120,6 +122,153 @@ Let's try out using `resources` in our TaskList project to see what will happen.
 
+## Comprehension Questions
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: checkbox
+* id: 22b7c5a4-3425-4741-9ff1-15036e97730d
+* title: Valid Routes Definitions
+* points: 1
+* topics: rails, routes
+
+##### !question
+
+Which of the following are valid route definitions?
+
+##### !end-question
+
+##### !options
+
+* `resources :pandas`
+* `resources :pandas, only: [:show]`
+* `resources :pandas, except: [:index, :edit, :update]`
+* `resources :pandas, only: [:index, :show]`
+
+##### !end-options
+
+##### !answer
+
+* `resources :pandas`
+* `resources :pandas, only: [:show]`
+* `resources :pandas, except: [:index, :edit, :update]`
+* `resources :pandas, only: [:index, :show]`
+
+
+##### !end-answer
+
+<!-- other optional sections -->
+<!-- !hint - !end-hint (markdown, users can see after a failed attempt) -->
+<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+##### !explanation
+
+Each of these definitions is correct! 
+
+##### !end-explanation
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: short-answer
+* id: b0b89a70-c309-43c9-b7ea-b268210af3ea
+* title: Override the root path
+* points: 1
+* topics: rails, routes
+
+##### !question
+
+How can you make the homepage of a rails app (i.e. http://localhost:3000/) point to the `TasksController` and `index` action?
+
+##### !end-question
+
+##### !placeholder
+
+How to override the root path?
+
+##### !end-placeholder
+
+##### !answer
+
+/\s*root\s*(to\:)?\s* [\'\"]tasks\#index[\'\"]\s*/
+
+##### !end-answer
+
+<!-- other optional sections -->
+##### !hint
+
+You can look above at a code snippet for an example.
+
+##### !end-hint
+<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+##### !explanation
+
+You can make the home or "root" page of the rails site go to a specific controller with:
+
+`root to: "tasks#index"`, you can also do `root "tasks#index"`
+
+##### !end-explanation
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: short-answer
+* id: 48c2bad7-4e4b-4236-8e19-30abb3db8e6f
+* title: Why manual routes then?
+* points: 1
+* topics: rails, routes
+
+##### !question
+
+So... if we have `resources`, do we need manual routes like the one below?
+
+```ruby
+patch mark_complete_path(:id) to: "tasks#mark_complete"
+```
+
+If so, why?
+
+##### !end-question
+
+##### !placeholder
+
+Do we need manual routes?
+
+##### !end-placeholder
```

Comment:
This is a GREAT question! I want to keep it!

Just something that crossed my mind: I didn't actually hear this material covered in the video... I might've missed it. Was it covered in the video? Even if it wasn't, this is a good question to make them think, but yeah :)

Also, I think this lesson ends up using "custom routes" and "manual routes" interchangeably.

My read says that this question might feel more consistent with the rest of the language if we stick to "custom"

This isn't that big of a deal, but I'm adding a suggestion for an easy commit just in case you want to go for it

```suggestion
##### !question

So... if we have `resources`, do we need custom routes like the one below?

```ruby
patch mark_complete_path(:id) to: "tasks#mark_complete"
```

If so, why?

##### !end-question

##### !placeholder

Do we need custom routes?

##### !end-placeholder
```

Reply:
It wasn't covered in the video, but it is in the text and it connects to TaskList... an oversight.  

However I will change it to talk about `custom` routes instead of manual, to keep consistent.